### PR TITLE
Monkeypatch for Llama 3.2-Vision

### PR DESCRIPTION
**Pull request by tyler-romero**
## Summary
Add monkeypatch to support [Llama 3.2-Vision](https://github.com/huggingface/transformers/pull/33703/files#diff-d804e851851cdebeb8048938f1f8beec1cfa78bf7b1f06af86faa450f9d18def) models.

## Details
Llama 3.2-Vision is a multimodal model. It is also only available in transformers>=4.45.0.

Torchvision is required to run the multimodal tests for Llama 3.2-Vision (the image processor requires it).

## Testing Done

- Hardware Type: RTX 4090
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by tyler-romero** [comment]
TODO:
1. Patch the vision tower as well
2. no-logits test

**Comment by tyler-romero** [comment]
All tests passing except for unrelated gemma2 tests:
```
====================================================================== short test summary info ======================================================================
FAILED test/convergence/test_mini_models.py::test_mini_model[mini_gemma2-32-0.0001-dtype4-1e-08-1e-05-0.005-1e-05-0.005-1e-05] - AssertionError: Number of mismatched elements: 31
FAILED test/convergence/test_mini_models.py::test_mini_model[mini_gemma2-32-0.0001-dtype5-0.001-0.01-0.1-0.01-0.01-0.01] - AssertionError: Number of mismatched elements: 12
FAILED test/convergence/test_mini_models_no_logits.py::test_mini_model[mini_gemma2-32-0.0001-dtype16-1e-08-0.0001-0.005-1e-05-0.005-1e-05] - AssertionError: Number of mismatched elements: 31
FAILED test/convergence/test_mini_models_no_logits.py::test_mini_model[mini_gemma2-32-0.0001-dtype17-0.001-0.01-0.1-0.01-0.01-0.01] - AssertionError: Number of mismatched elements: 12
======================================================= 4 failed, 32 passed, 9 warnings in 264.08s (0:04:24) ========================================================
```

File: test/transformers/test_monkey_patch.py [hunk 1]
```diff
@@ -229,10 +234,133 @@ def test_apply_liger_kernel_to_instance_for_llama():
             assert isinstance(layer.post_attention_layernorm, LigerRMSNorm)
 
 
+def test_apply_liger_kernel_to_instance_for_mllama_for_conditional_generation():
+    # Ensure any monkey patching is cleaned up for subsequent tests
+    with patch("transformers.models.mllama.modeling_mllama"):
```
Comment (tyler-romero) [inline test/transformers/test_monkey_patch.py:239]: FWIW, this method of trying to clean up monkeypatching doesnt work. Try running two identical test from this file back to back.

**Comment by tyler-romero** [comment]
Wanted to add support for LayerNorm in the vision tower as well, but it is hard to patch the way it is used in modeling_mllama.py - a patch would end up changing torch.nn globally.

**Comment by tyler-romero** [comment]
Just need someone to set a HF Token that has accepted the llama license in the CICD env:
https://discord.com/channels/1189498204333543425/1275130785933951039/1290378722653896744
```
>           raise EnvironmentError(
                "You are trying to access a gated repo.\nMake sure to have access to it at "
                f"[https://huggingface.co/{path_or_repo_id}.\n{str(e)}](https://huggingface.co/%7Bpath_or_repo_id%7D./n%7Bstr(e)%7D)"
            ) from e
E           OSError: You are trying to access a gated repo.
E           Make sure to have access to it at https://huggingface.co/meta-llama/Llama-3.2-11B-Vision-Instruct.
E           401 Client Error. (Request ID: Root=1-66fc5343-253e99011960556f10a26970;1bfe5fb8-1fa8-4ff0-a21e-28041d5b6752)
E           
E           Cannot access gated repo for url https://huggingface.co/meta-llama/Llama-3.2-11B-Vision-Instruct/resolve/main/config.json.
E           Access to model meta-llama/Llama-3.2-11B-Vision-Instruct is restricted. You must have access to it and be authenticated to access it. Please log in.
```

**Comment by shivam15s** [comment]
Added HF hub secret to the repo. Waiting for the HF account to get approved for llama 3.2 models

**Comment by tyler-romero** [comment]
Thanks, for some reason I'm still seeing the same issue - could you try triggering a cicd run on this branch?

**Comment by shivam15s** [comment]
Let me get back. Still figuring out how to configure the gpu to use the access token.

**Comment by shivam15s** [comment]
Seems like using gated stuff for tests isn't a good idea. I'm looking into using fake/mirrored models/processors for the tests

**Comment by tyler-romero** [comment]
Any update @shivam15s ?

**Comment by shivam15s** [comment]
Hey @tyler-romero, created a PR on your branch. Let me know if the proposed solution makes sense

**Comment by shivam15s** [comment]
Seems like a convergence test fails because of a precision issue.
Could you please look into it? Thanks

**Comment by tyler-romero** [comment]
Now that you made custom tokenizers for the multimodal tests, I could reduce the vocab sizes of the mini-models (to be in-line with the vocab sizes of the other mini-models), and the convergence tests are passing now.

**Review by shivam15s — APPROVED** [review]
Great work, lgtm

**Comment by tyler-romero** [comment]
Sounds good, feel free to merge (only collaborators/linkedin employees can merge) - and please add yourself as a contributor to this change if GitHub doesn't do that automatically.

**Comment by ByronHsu** [comment]
Thank for both for driving this!!

**Pull request merged**